core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/OtherSignUpOptionSelectionViewModel.kt

```diff
@@ -115,47 +115,80 @@ class OtherSignUpOptionSelectionViewModel(
-                    savePerson,
-                    _uiState.value.personPicture
-                )
-            }
-            if (passkeyCreated == null) {
-                snackDispatcher.showSnackBar(Snack(message = systemImpl.getString(MR.strings.sorry_something_went_wrong)))
-                Napier.e { "Error occurred during creating passkey" }
-                return@launch
-            }
-            if (isParent) {
-                navController.navigate(
-                    AddChildProfilesViewModel.DEST_NAME,
-                    args = buildMap {
-                        put(ARG_NEXT, nextDestination)
-                        putAllFromSavedStateIfPresent(REGISTRATION_ARGS_TO_PASS)
-                        putFromSavedStateIfPresent(ARG_NEXT)
+            when(passkeyCreated){
+                is CreatePasskeyUseCase.CreatePasskeyResult -> {
+                    viewModelScope.launch {
+                        onReceivedCreatePasskeyResult(
+                            passkeyCreated.authenticationResponseJSON,
+                            savePerson
+                        )
                     }
-                )
-
-            } else {
-                enrollToCourseFromInviteUid(savePerson.personUid)
-                val goOptions = UstadMobileSystemCommon.UstadGoOptions(clearStack = true)
-                Napier.d { "AddSignUpPresenter: go to next destination: $nextDestination" }
-                navController.navigateToViewUri(
-                    nextDestination.appendSelectedAccount(
-                        savePerson.personUid,
-                        LearningSpace(accountManager.activeLearningSpace.url)
-                    ),
-                    goOptions
-                )
 
+                }
+                is CreatePasskeyUseCase.Error ->{
+                    _uiState.update { prev ->
+                        prev.copy(
+                            errorText = passkeyCreated.message,
+                        )
+                    }
+                }
+                is CreatePasskeyUseCase.UserCanceledResult ->{
+                    //do nothing
+                }
+
+                null -> {
+                    //do nothing
+                }
             }
+
         }
 
-       // navController.popBackStack(SignUpViewModel.DEST_NAME,false)
 
     }
+
+
+    private suspend fun onReceivedCreatePasskeyResult(
+        passkeyCreated: AuthenticationResponseJSON,
+        savePerson: Person
+    ) {
+        passkeyCreated?.let {
+            accountManager.registerWithPasskey(
+                serverUrl,
+                it,
+                savePerson,
+                _uiState.value.personPicture
+            )
+        }
+        if (passkeyCreated == null) {
+            snackDispatcher.showSnackBar(Snack(message = systemImpl.getString(MR.strings.sorry_something_went_wrong)))
+            Napier.e { "Error occurred during creating passkey" }
+            return
+        }
+        if (isParent) {
```

Comment:
These changes appear unrelated to the issues this pull request  is supposed to be addressing